provider/resource_keycloak_realm_keystore_aes_generated.go

```diff
@@ -0,0 +1,174 @@
+			},
+			"priority": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Priority for the provider",
+			},
+			"secret_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntInSlice(keycloakRealmKeystoreAesGeneratedSize),
+				Default:      16,
+				Description:  "Size in bytes for the generated AES Key. Size 16 is for AES-128, Size 24 for AES-192 and Size 32 for AES-256. WARN: Bigger keys then 128 bits are not allowed on some JDK implementations",
+			},
+		},
+	}
+}
+
+func getRealmKeystoreAesGeneratedFromData(data *schema.ResourceData) (*keycloak.RealmKeystoreAesGenerated, error) {
+	keystore := &keycloak.RealmKeystoreAesGenerated{
+		Id:      data.Id(),
+		Name:    data.Get("name").(string),
+		RealmId: data.Get("realm_id").(string),
+
+		Active:     data.Get("active").(bool),
+		Enabled:    data.Get("enabled").(bool),
+		Priority:   data.Get("priority").(int),
+		SecretSize: data.Get("secret_size").(int),
+	}
+
+	return keystore, nil
+}
+
+func setRealmKeystoreAesGeneratedData(data *schema.ResourceData, realmKey *keycloak.RealmKeystoreAesGenerated) error {
+	data.SetId(realmKey.Id)
+
+	data.Set("name", realmKey.Name)
+	data.Set("realm_id", realmKey.RealmId)
+
+	data.Set("active", realmKey.Active)
+	data.Set("enabled", realmKey.Enabled)
+	data.Set("priority", realmKey.Priority)
+	data.Set("secret_size", realmKey.SecretSize)
+
+	return nil
+}
+
+func resourceKeycloakRealmKeystoreAesGeneratedCreate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmKey, err := getRealmKeystoreAesGeneratedFromData(data)
+	if err != nil {
+		return err
+	}
+
+	err = keycloakClient.NewRealmKeystoreAesGenerated(realmKey)
+	if err != nil {
+		return err
+	}
+
+	err = setRealmKeystoreAesGeneratedData(data, realmKey)
+	if err != nil {
+		return err
+	}
+
+	return resourceKeycloakRealmKeystoreAesGeneratedRead(data, meta)
+}
+
+func resourceKeycloakRealmKeystoreAesGeneratedRead(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+	id := data.Id()
+
+	realmKey, err := keycloakClient.GetRealmKeystoreAesGenerated(realmId, id)
+	if err != nil {
+		return handleNotFoundError(err, data)
+	}
+
+	err = setRealmKeystoreAesGeneratedData(data, realmKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKeycloakRealmKeystoreAesGeneratedUpdate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmKey, err := getRealmKeystoreAesGeneratedFromData(data)
+	if err != nil {
+		return err
+	}
+
+	err = keycloakClient.UpdateRealmKeystoreAesGenerated(realmKey)
+	if err != nil {
+		return err
+	}
+
+	err = setRealmKeystoreAesGeneratedData(data, realmKey)
+	if err != nil {
+		return err
+	}
+
+	return keycloakClient.UpdateRealmKeystoreAesGenerated(realmKey)
```

Comment:
is there a reason we're invoking `keycloakClient.UpdateRealmKeystoreAesGenerated` twice here.  all of these new resources appear to do this as well.